sky/provision/vast/utils.py

```diff
@@ -0,0 +1,103 @@
+# pylint: disable=assignment-from-no-return
+"""Vast library wrapper for SkyPilot."""
+from typing import Any, Dict, List, Optional
+
+from sky import sky_logging
+from sky.adaptors import vast
+
+logger = sky_logging.init_logger(__name__)
+
+
+def list_instances() -> Dict[str, Dict[str, Any]]:
+    """Lists instances associated with API key."""
+    instances = vast.vast().show_instances()
+
+    instance_dict: Dict[str, Dict[str, Any]] = {}
+    for instance in instances:
+        instance['id'] = str(instance['id'])
+        info = instance.copy()
+
+        if isinstance(instance['actual_status'], str):
+            info['status'] = instance['actual_status'].upper()
+        else:
+            info['status'] = 'UNKNOWN'
+        info['name'] = instance['label']
+
+        instance_dict[instance['id']] = info
+
+    return instance_dict
+
+
+def launch(name: str, instance_type: str, region: str, disk_size: int,
+           image_name: str, ports: Optional[List[int]], public_key: str) -> str:
+    """Launches an instance with the given parameters.
+
+    Converts the instance_type to the Vast GPU name, finds the specs for the
+    GPU, and launches the instance.
+    """
+    del ports
+    del public_key
+
+    gpu_name = instance_type.split('-')[1].replace('_', ' ')
+    num_gpus = int(instance_type.split('-')[0].replace('x', ''))
+
+    query = ' '.join([
+        f'geolocation="{region[-2:]}"',
+        f'disk_space>={disk_size}',
```

Comment:
Why using `>=` here? If it is because sth like vast only supports power-of-2 disk size, a comment would be helpful :)

Reply:
disk size is non-uniform on vast. Some providers can offer say 1000GB and another might offer 1020GB or 1015GB. This is to not exclude those otherwise suitable machines from consideration if a user specified 1000GB.

In a quick survey I did of 750 machines on our network, 83% of them offered unique disk capacities.  For instance, we had a 1006GB and a 1008GB machine available.

Reply:
see https://github.com/kristopolous/skypilot/commit/6a6c21579fa1b5b88ab1c74053bd2295fa7c2ab1

Reply:
Also, can we check the disk size after getting the offering and print out some warnings if the disk size is not exactly match?

Reply:
This is basically a guarantee. I really don't want to go through a month of back and forth on this. What do you want it to say?

Reply:
sth like `The disk size {xx} GB is not exactly matched the requested size {yy} GB. It is possible to charge extra cost on disk.`